### PR TITLE
[ML] Python module version updates

### PR DESCRIPTION
**Pull request by droberts195**
As recommended by Snyk

### CODE REVIEW

**Comment by valeriy42** [comment]
I close this pull request. When #2595 is merged, we don't need to update Python dependencies anymore.

File: jupyter/requirements.txt [hunk 1]
```diff
@@ -11,13 +11,14 @@ jupyter             ==1.0.0
...
 matplotlib          ==3.3.4
 nbsmoke             ==0.5.0
-numpy               >=1.21.0
+numpy               >=1.22.2
 pandas              >=1.3
 pathlib2            ==2.3.5
 plotly              ==5.3.1
 sacred              ==0.8.2
-scikit-learn        ==0.24.1
+scikit-learn        ==0.24.2
```
Comment (valeriy42) [inline jupyter/requirements.txt:19]:
```suggestion
scikit-learn        ==1.3
```

**Review by valeriy42 — APPROVED** [review]
The older version of scikit-learn (0.x) seem to have problem with installation using newer Python interpreters. We can switch to the 1.x version brunch. 

Other than this, it's LGTM.

**Comment by droberts195** [comment]
> The older version of scikit-learn (0.x) seem to have problem with installation using newer Python interpreters.

Thanks for testing it @valeriy42.

**Pull request merged**